apps/src/code-studio/pd/workshop_enrollment/enroll_form.tsx

```diff
@@ -352,102 +362,129 @@ export default function EnrollForm(props: EnrollFormProps) {
         }
       );
       setIsSubmitting(false);
-      if (response.ok) {
```

Comment:
Why remove this check??

Reply:
ah, this was tricky! any validation errors are returned in the response json, so it needs to be parsed regardless of the response status and handled in `onSubmissionComplete`. this was a bug I accidentally introduced when I first refactored the component